Assets/Packages/com.studio23.ss2.objectivesystem/Runtime/Core/ObjectiveBase.cs

```diff
@@ -0,0 +1,322 @@
+using System;
+using System.Collections.Generic;
+using NaughtyAttributes;
+using Newtonsoft.Json;
+using Studio23.SS2.InventorySystem.Data;
+using Studio23.SS2.ObjectiveSystem.Data;
+using UnityEngine;
+using UnityEngine.Serialization;
+
+namespace Studio23.SS2.ObjectiveSystem.Core
+{
+    [CreateAssetMenu(menuName = "Studio-23/Objective System/Basic Objective", fileName = "Basic Objective")]
+    [Serializable]
+    public class ObjectiveBase : ItemBase
+    {
+        // we do not actually need to serialize this in editor
+        // this will be different based on save data.
+        // we prevent serialization to avoid merge conflicts
+        // This is manually serialized when saved as needed.
+        [ShowNonSerializedField] private ObjectiveState _state;
```

Comment:
I think we should use [jsonignore] attribute to make sure these are not serialized?